wikimedia/wikipedia-ios
# Native Editor - Add list support

**Phabricator:** 
https://phabricator.wikimedia.org/T348090

### Notes
Implements syntax highlighting and button actions for list buttons.

### Test Steps
1. On Staging app, go to an article.
2. On a new line, type one or more asterisks (*), then some letters. Confirm asterisks are orange.
3. On another new line, type one or more hashes (#), then some letters. Confirm hashes are orange.
4. Change theme in popover, confirm orange changes as expected in each theme.
5. Type text on a new line, confirm list buttons (bullets, numbers, indentions) are not selected.
6. Tap bullet button. Confirm a * is inserted at the front of the line. Confirm increase indent button is enabled, but decrease is still disabled.
7. Tap increase indention button. Confirm another * is inserted at the front of the line. Confirm decrease indention button is now enabled.
8. Tap decrease indent button. Confirm * from previous step is removed.
9. Tap bullet button to deselect. Confirm original * is removed.
10. Repeat steps 6 - 9 for numbered lists.
